src/main/java/baseball/models/Baseball.java

```diff
@@ -0,0 +1,41 @@
+package baseball.models;
+
+import java.util.List;
+import java.util.stream.IntStream;
+
+public class Baseball {
+    public static final int BALL_SIZE = 3;
+    private Integer strikeCount = 0;
+    private Integer ballCount = 0;
+
+    public void playBaseball(List<Integer> playerBall, List<Integer> computerBall) {
+        countStrike(playerBall, computerBall);
+        countBall(playerBall, computerBall);
+    }
+
+    public Integer getStrikeCount() {
+        return strikeCount;
+    }
+
+    public Integer getBallCount() {
+        return ballCount;
+    }
+
+    public boolean isBaseBallEnd() {
+        return strikeCount < BALL_SIZE;
+    }
+
+    private void countStrike(List<Integer> playerBall, List<Integer> computerBall) {
+        long strikeCount = IntStream.range(0, playerBall.size())
+                .filter(idx -> playerBall.get(idx).equals(computerBall.get(idx)))
+                .count();
+        this.strikeCount = Long.valueOf(strikeCount).intValue();
+    }
+
+    private void countBall(List<Integer> playerBall, List<Integer> computerBall) {
+        long totalCount = playerBall.stream()
+                .filter(computerBall::contains)
+                .count();
+        this.ballCount = Long.valueOf(totalCount).intValue() - strikeCount;
```

Comment:
```suggestion
        int totalCount = Math.toIntExact(playerBall.stream()
                .filter(computerBall::contains)
                .count());
        this.ballCount = totalCount - strikeCount;
    }
```

이런식으로 하면 더 깔끔하겠네요 :)

Reply:
https://github.com/programmers-lecture/java-baseball/pull/3/commits/ca004fd1e6176dcfedb88e8483dc19ce70634fa0
참고해서 수정했습니다!